mangaki/mangaki/admin.py

```diff
@@ -149,103 +268,11 @@ def make_sfw(self, request, queryset):
-                work_to_delete.delete()
-            kept_work.save()
-            return None
-
-        do_not_compare = ['sum_ratings', 'nb_ratings', 'nb_likes', 'nb_dislikes', 'controversy']
-        priority = {
-            MergeType.CHOICE_REQUIRED: 2,
-            MergeType.JUST_CONFIRM: 1,
-            MergeType.INFO_ONLY: 0
-        }
-        row_color = {
-            MergeType.CHOICE_REQUIRED: 'red',
-            MergeType.JUST_CONFIRM: 'green',
-            MergeType.INFO_ONLY: 'black'
-        }
-        merge_type = {}
-        rows = defaultdict(list)
-        for work in queryset.values():
-            for field in work:
-                rows[field].append(work[field])
-        fields_to_choose = []
-        template_rows = []
-        for field in rows:
-            choices = rows[field]
-            suggested = None
-            if field in do_not_compare:
-                merge_type[field] = MergeType.INFO_ONLY
-            elif all(choice == choices[0] for choice in choices):  # All equal
-                merge_type[field] = MergeType.JUST_CONFIRM
-                suggested = choices[0]
-            elif sum(not choice or choice == 'Inconnu' for choice in choices) == len(choices) - 1:  # All empty but one
-                merge_type[field] = MergeType.JUST_CONFIRM
-                suggested = [choice for choice in choices if choice and choice != 'Inconnu'][0]  # Remaining one
-            else:
-                merge_type[field] = MergeType.CHOICE_REQUIRED
-            template_rows.append({
-                'field': field,
-                'choices': choices,
-                'merge_type': merge_type[field],
-                'suggested': suggested,
-                'color': row_color[merge_type[field]],
-            })
-            if field != 'id' and merge_type[field] != MergeType.INFO_ONLY:
-                fields_to_choose.append(field)
-        template_rows.sort(key=lambda row: priority[row['merge_type']], reverse=True)
-
-        rating_samples = []
-        for work in queryset:
-            rating_samples.append(
-                (Rating.objects.filter(work=work).count(),
-                 Rating.objects.filter(work=work)[:10])
-            )
-
-        context = {
-            'fields_to_choose': ','.join(fields_to_choose),
-            'works_to_merge': queryset,
-            'template_rows': template_rows,
-            'rating_samples': rating_samples,
-            'queryset': queryset,
-            'opts': self.model._meta,
-            'action_checkbox_name': helpers.ACTION_CHECKBOX_NAME
-        }
-        return TemplateResponse(request, 'admin/merge_selected_confirmation.html', context)
+        response = merge_works(request, queryset, queryset, from_cluster=False)
+        if response is None:
+            self.message_user(request, "La fusion %s a bien été effectuée." % '-'.join(str(work.id) for work in queryset))
```

Comment:
I'm not sure it makes that much sense to display the IDs here. Here is how I would (mostly) write this function:

```python
from django.utils.html import format_html  # At top of file :-)

num_to_merge = len(queryset)  # Could possibly be returned by merge_works
response = merge_works(request, queryset, queryset, from_cluster=False)
if response is None:
    final_work = queryset.get()  # Should definitely be returned by merge_works 
    self.message_user(request, format_html(
        "La fusion de {:d} œuvres vers <a href="{:s}">{:s}</a> a été effectuée." % (
            num_to_merge, final_work.get_absolute_url(), final_work.title)))
return response
```